whipper/common/mbngs.py

```diff
@@ -161,10 +161,78 @@ def _getWorks(recording):
     if 'work-relation-list' in recording:
         for work in recording['work-relation-list']:
             if work['type-id'] in valid_work_rel_types:
-                works.append(work['work']['id'])
+                works.append(work['work'])
     return works
 
 
+def _getWorksId(works):
+    """Get work ids from "performance of" works"""
+    ids = []
+    for work in works:
+        ids.append(work['id'])
+    return ids
+
+
+def _getComposers(works):
+    """
+    Extract composer(s) from works' artist-relation-list into a list
+
+    :rtype: list
+    """
+    composers = []
+    valid_artist_rel_type = [
+        'd59d99ea-23d4-4a80-b066-edca32ee158f',  # "Composer"
+    ]
+    for work in works:
+        if 'artist-relation-list' in work:
+            for artist_relation in work['artist-relation-list']:
+                if artist_relation['type-id'] in valid_artist_rel_type:
+                    composerName = artist_relation['artist']['name']
+                    composers.append(composerName)
+    return composers
+
+
+def _getPerformers(artist_relation_list):
+    """
+    Extract composers from recordings' artist-relation-list
+
+    :rtype: list
+    """
+    performers = []
+    valid_type_id = [
+        '59054b12-01ac-43ee-a618-285fd397e461',  # "Instruments"
+        '0fdbe3c6-7700-4a31-ae54-b53f06ae1cfa',  # "Vocals"
+        '628a9658-f54c-4142-b0c0-95f031b544da'   # "Performers"
+    ]
+    common_attribute_name = [
+        'guest', 'additional', 'solo'
+    ]
+    for artist_relation in artist_relation_list:
+        if artist_relation['type-id'] in valid_type_id:
+            type = artist_relation['type']
+            name = artist_relation['artist']['name']
+
+            if 'attribute-list' in artist_relation:
+                prefix = ''  # additional guest solo
+                others = []  # types of vocal, instruments
+                for attribute in artist_relation['attribute-list']:
+                    if attribute in common_attribute_name:
+                        # this works since attributes' order is alphabetical
+                        # which will correctly match the given standard order
+                        # see: https://musicbrainz.org/relationship/eb10f8a0-0f4c-4dce-aa47-87bcb2bc42f3  # noqa E501
+                        prefix += attribute + ' '
+                    else:
+                        others.append(attribute)
+                # if "guest performer: artist", `suffix` is `type` by default
+                # else "guest solo piano, organ : artist"
+                suffix = type if len(others) == 0 else ', '.join(others)
+                type = prefix + suffix
+
+            performer = "%s: %s" % (type, name)
+            performers.append(performer)
+    return performers
```

Comment:
(And same question about same name appearing multiple times. :))

Reply:
I mean for this one since the code is not looping through multiple works it should be fine just like in the test case Lenny Krevitz already appears twice.

Reply:
What do you think about this, @Freso does this need modification too.

Reply:
@Freso ping 😉

Reply:
It’s probably fine. If it turns out there are issues with this, that can be dealt with later. I think this is good for now. :)